engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -1264,12 +1264,12 @@ public Operator loadOperator(PTOperator oper)
    */
   private Checkpoint getActivationCheckpoint(PTOperator oper)
   {
-    if (oper.recoveryCheckpoint == null && oper.checkpoints.isEmpty()) {
+    if ((oper.recoveryCheckpoint == null || oper.recoveryCheckpoint == Checkpoint.INITIAL_CHECKPOINT) && oper.checkpoints.isEmpty()) {
```

Comment:
It was supposed to be noop for an already initialized operator. Why is recoveryCheckpoint assigned in your case?